src/combat.cpp

```diff
@@ -780,35 +816,26 @@ void Combat::doCombatHealth(Creature* caster, Creature* target, CombatDamage& da
 			addDistanceEffect(caster, caster->getPosition(), target->getPosition(), params.distanceEffect);
 		}
 
-		CombatHealthFunc(caster, target, params, &damage);
+		func(caster, target, params, damage);
```

Comment:
check

Reply:
I've made it generic, 1 function that receives the func callback instead of 5 functions doing the same thing with different callbacks.